operators/src/processing/vector_join/equi_data_join.rs

```diff
@@ -0,0 +1,401 @@
+use std::collections::HashSet;
+use std::marker::PhantomData;
+
+use futures::stream;
+use futures::stream::BoxStream;
+use futures::StreamExt;
+
+use geoengine_datatypes::collections::{
+    BuilderProvider, DataCollection, FeatureCollection, FeatureCollectionInfos,
+    FeatureCollectionModifications, FeatureCollectionRowBuilder, GeoFeatureCollectionRowBuilder,
+    IntoGeometryIterator,
+};
+use geoengine_datatypes::primitives::{DataRef, FeatureDataRef, Geometry};
+use geoengine_datatypes::util::arrow::ArrowTyped;
+
+use crate::adapters::FeatureCollectionChunkMerger;
+use crate::engine::{QueryContext, QueryProcessor, QueryRectangle, VectorQueryProcessor};
+use crate::error::Error;
+use crate::util::Result;
+
+/// Implements a left inner join between a `GeoFeatureCollection` stream and a `DataCollection` stream.
+pub struct EquiLeftJoinProcessor<G> {
+    vector_type: PhantomData<FeatureCollection<G>>,
```

Comment:
do you need it? isn't it  bound by `left_processor`?

Reply:
27ee0c0